### PR TITLE
THORN-2195: Use the correct TCCL when health checks are obtained.

### PR DESCRIPTION
**Pull request by mkouba**

### CODE REVIEW

File: fractions/microprofile/microprofile-health/src/main/java/org/wildfly/swarm/microprofile/health/api/Monitor.java [hunk 1]
```diff
@@ -54,6 +54,12 @@ static Monitor lookup() throws NamingException {
 
     Object getHealthReporter();
 
+    void unregisterContextClassLoader();
+
+    void registerContextClassLoader(ClassLoader classLoader);
+
+    ClassLoader getContextClassLoader();
+
```
Comment (Ladicek) [inline fractions/microprofile/microprofile-health/src/main/java/org/wildfly/swarm/microprofile/health/api/Monitor.java:62]: Just wondering if we want to make this part of the public API?

**Comment by Ladicek** [comment]
Otherwise LGTM.

File: fractions/microprofile/microprofile-health/src/main/java/org/wildfly/swarm/microprofile/health/api/Monitor.java (hunk 1 above)
Comment (sberyozkin) [reply fractions/microprofile/microprofile-health/src/main/java/org/wildfly/swarm/microprofile/health/api/Monitor.java:62]: That should be fine, all that class is there for is to facilitate the interaction between various microprofile health parts as far as I understand, not really meant for the public use. May be it should be noted in the docs somehow

Comment (Ladicek) [reply fractions/microprofile/microprofile-health/src/main/java/org/wildfly/swarm/microprofile/health/api/Monitor.java:62]: I agree it's not _meant_ for public use, but IIUC, it _is_ exposed publicly.

Comment (mkouba) [reply fractions/microprofile/microprofile-health/src/main/java/org/wildfly/swarm/microprofile/health/api/Monitor.java:62]: Well, all public classes/methods/etc. are exposed and can be considered a part of the API ;-)

Comment (sberyozkin) [reply fractions/microprofile/microprofile-health/src/main/java/org/wildfly/swarm/microprofile/health/api/Monitor.java:62]: I don't mind either way :-). I recall looking at that class and I'm not sure any of its methods could be meaningfully used by the non Thorntail code...

Comment (Ladicek) [reply fractions/microprofile/microprofile-health/src/main/java/org/wildfly/swarm/microprofile/health/api/Monitor.java:62]: We actually have a delineation between public API and private details, e.g. the `runtime` packages are not considered public API. The delineation is defined by the script that generates Javadoc :-)

Comment (Ladicek) [reply fractions/microprofile/microprofile-health/src/main/java/org/wildfly/swarm/microprofile/health/api/Monitor.java:62]: But @sberyozkin is right, this API isn't really all that useful and is "here be dragons" already :-)

**Review by Ladicek — APPROVED** [review]

**Pull request merged**